libbeat/esleg/eslegclient/connection.go

```diff
@@ -394,19 +402,34 @@ func (conn *Connection) GetVersion() libversion.V {
 	return conn.version
 }
 
+// IsServerless returns true if we're connected to a serverless ES instance
+func (conn *Connection) IsServerless() bool {
+	// make sure we've initialized the version state first
+	_ = conn.GetVersion()
+	return conn.isServerless
+}
+
 func (conn *Connection) getVersion() error {
-	versionString, err := conn.Ping()
+	versionData, err := conn.Ping()
 	if err != nil {
 		return err
 	}
 
-	if v, err := libversion.New(versionString); err != nil {
-		conn.log.Errorf("Invalid version from Elasticsearch: %v", versionString)
+	if v, err := libversion.New(versionData.Version.Number); err != nil {
+		conn.log.Errorf("Invalid version from Elasticsearch: %v", versionData.Version.Number)
 		conn.version = libversion.V{}
 	} else {
 		conn.version = *v
 	}
 
+	if versionData.Version.BuildFlavor == "serverless" {
+		conn.isServerless = true
+	} else if versionData.Version.BuildFlavor == "default" {
+		conn.isServerless = false
```

Comment:
Can this ever actually change, or is this just defensive programming? If the latter can you add a comment saying something like "this should already be false but let's be extra careful"?